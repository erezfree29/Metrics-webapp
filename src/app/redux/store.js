/* eslint-disable import/prefer-default-export */
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import logger from 'redux-logger';

import laLigaReducer from './laliga';

import premierReducer from './premer';

import seriaReducer from './seriaA';

import bundesLigaReducer from './bumdesliga';

const reducer = combineReducers({
  laLigaReducer,
  premierReducer,
  seriaReducer,
  bundesLigaReducer,
});

export const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);
