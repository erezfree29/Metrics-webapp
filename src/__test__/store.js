/* eslint-disable import/prefer-default-export */
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import logger from 'redux-logger';

import laLigaReducer from '../app/redux/laliga';

import premierReducer from '../app/redux/premer';

import seriaReducer from '../app/redux/seriaA';

const reducer = combineReducers({
  laLigaReducer,
  premierReducer,
  seriaReducer,
});

export const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);
