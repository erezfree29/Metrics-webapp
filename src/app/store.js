/* eslint-disable import/prefer-default-export */
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import logger from 'redux-logger';

const reducer = combineReducers({
});

export const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);
