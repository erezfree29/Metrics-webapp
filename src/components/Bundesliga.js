/* eslint-disable no-unused-vars */
/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable react/jsx-key */

import { NavLink } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import React, { useEffect } from 'react';

import { updateBLiga } from '../app/redux/bumdesliga';

function Bundesliga() {
  const dispatch = useDispatch();
  const bmatches = useSelector((state) => state.bundesLigaReducer);
  const fetchItems = async () => {
    const data = await fetch('https://www.scorebat.com/video-api/v3/');
    const matchesJason = await data.json();
    const bundesMatches = matchesJason.response.filter((match) => match.competition === 'GERMANY: Bundesliga');
    dispatch(updateBLiga(bundesMatches));
  };

  useEffect(() => {
    if (bmatches.length === 0) {
      fetchItems();
    }
  }, []);

  return (
    <>
      <nav className="navbar">
        <div className="link">
          <NavLink
            to="/"
          >
            <div className="back"><i className="fas fa-angle-left" /></div>
          </NavLink>
        </div>
        <div className="views">League matches</div>
      </nav>
      <div className="hand move_down bhand"><i className="fas fa-hand-point-up" /></div>
      <div className="breakdwon">League team Breakdown 2021</div>
      <div className="bundes_liga">
        <div className="league_top">
          <div className="lename name"> Bundesliga</div>
          <img className="image_top" src="https://res.cloudinary.com/erezfriemagor/image/upload/v1632348556/27160_imgw968.png" alt="premier" />
        </div>
        <table className=" table table-borderless table-striped">
          <thead>
            <tr>
              <th scope="col" />
            </tr>
          </thead>
          <tbody>
            {bmatches.map((match) => (
              <tr>
                <td>
                  <a href={match.matchviewUrl}>
                    <div className="mcard">
                      <div>{match.title}</div>
                      <div className="mdate">
                        <div className="dword">Date</div>
                        <div>{match.date.slice(0, 10)}</div>
                        <div><i className="fas fa-arrow-circle-right" /></div>
                      </div>
                    </div>
                  </a>
                </td>
              </tr>

            ))}
          </tbody>
        </table>
      </div>

    </>

  );
}
export default Bundesliga;
