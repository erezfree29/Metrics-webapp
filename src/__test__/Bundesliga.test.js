/**
 * @jest-environment jsdom
 */
import React from 'react';
import 'regenerator-runtime/runtime';
import { render } from '@testing-library/react';
// import App from '../App';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from './store';
import Bundesliga from './bundesliga';
import '@testing-library/jest-dom/extend-expect';

describe('check if all component are rendered', () => {
  it('renders page correctly', () => {
    const tree = render(<Provider store={store}><Router><Bundesliga /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
