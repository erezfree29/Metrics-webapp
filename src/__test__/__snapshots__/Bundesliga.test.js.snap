// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if all component are rendered renders page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="matches container"
      />
      <table
        class="table table-borderless"
      >
        <thead>
          <tr>
            <th />
            <th />
            <th />
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </body>,
  "container": <div>
    <div
      class="matches container"
    />
    <table
      class="table table-borderless"
    >
      <thead>
        <tr>
          <th />
          <th />
          <th />
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
